.github/workflows/python-test-modern.yml

```
@@ -0,0 +1,93 @@
+          cache: 'poetry'
+          
+      - name: Install dependencies
+        run: poetry install --no-interaction
+        
+      - name: Run test suite with coverage
+        run: |
+          poetry run pytest --cov=proxybroker --cov-report=term-missing --cov-report=xml --cov-fail-under=60
+          
+      - name: Upload coverage to Codecov
+        if: matrix.python-version == '3.11'
+        uses: codecov/codecov-action@v3
```

actionlint (1.7.7)

the runner of "codecov/codecov-action@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)